### ISSUE TITLE
More than 16 syncgroups cannot be advertised via BLE

### ISSUE THREAD

**Issue by azinman** [vanadium/issues#1404]
The current BLE implementation has a hard limit on 16 advertisements. As each syncgroup is advertised on its own, it means we can only have 16 syncgroups advertised. This is a problem for many apps such as todos as each todo list has its own syncgroup.

@asadovsky @asimshankar @ultrasaurus @afandria

**Comment by azinman** [comment]
Pending CL https://vanadium-review.googlesource.com/#/c/23864 disables BLE to avoid this problem.

**Comment by azinman** [comment]
Updated CL to ignore failure as long as 1 plugin succeeds -- BLE is re-enabled.

**Comment by ultrasaurus** [comment]
@afandria I would expect we would see the same issue on Android as well?  or is the implementation substantially different?

**Comment by asimshankar** [comment]
The limit of 16 comes from:
https://github.com/vanadium/go.ref/blob/60698e61821e1905605ff2cabf4c230d33989386/lib/discovery/plugins/ble/encoding.go#L36

It is somewhat arbitrary, but the general idea was that we need to limit the total amount of data available from the GATT server. IIRC, @jhahn21 noted that the actual limit varies depending on the hardware and the software stack and 16 was an unscientifically chosen arbitrary bound by us.

Long story short though: Yes, this issue affects all platforms, it is not Android or iOS specific.

**Comment by azinman** [comment]
Would it be possible implement something like a bloom filter for the syncgroup ids or advertisement data, instead of listing all the data? Or at least in the case for >16 advertisements?
